rust-src/wallet_library/src/proofs.rs

```diff
@@ -0,0 +1,134 @@
+        let mut rng = rand::thread_rng();
+        let challenge = Challenge::new(rng.gen());
+        let params = GlobalContext::generate("Test".into());
+        let cred_id_exp = constants::ArCurve::generate_scalar(&mut rng);
+        let cred_id = CredentialRegistrationID::from_exponent(&params, cred_id_exp);
+        let credential_statements = vec![CredentialStatement::Account {
+            network: Network::Testnet,
+            cred_id,
+            statement: vec![
+                AtomicStatement::AttributeInRange {
+                    statement: AttributeInRangeStatement {
+                        attribute_tag: 3.into(),
+                        lower:         Web3IdAttribute::Numeric(80),
+                        upper:         Web3IdAttribute::Numeric(1237),
+                        _phantom:      PhantomData,
+                    },
+                },
+                AtomicStatement::AttributeNotInSet {
+                    statement: AttributeNotInSetStatement {
+                        attribute_tag: 1u8.into(),
+                        set:           [
+                            Web3IdAttribute::String(constants::AttributeKind("ff".into())),
+                            Web3IdAttribute::String(constants::AttributeKind("aa".into())),
+                            Web3IdAttribute::String(constants::AttributeKind("zz".into())),
+                        ]
+                        .into_iter()
+                        .collect(),
+                        _phantom:      PhantomData,
+                    },
+                },
+            ],
+        }];
+        let request = Request::<constants::ArCurve, Web3IdAttribute> {
+            challenge,
+            credential_statements,
+        };
+        let global_context = read_global();
+        let mut values = BTreeMap::new();
+        values.insert(3.into(), Web3IdAttribute::Numeric(137));
+        values.insert(
+            1.into(),
+            Web3IdAttribute::String(constants::AttributeKind("xkcd".into())),
+        );
+        let mut randomness = BTreeMap::new();
+        for tag in values.keys() {
+            randomness.insert(
+                *tag,
+                pedersen_commitment::Randomness::<constants::ArCurve>::generate(&mut rng),
+            );
+        }
+        let secrets = OwnedCommitmentInputs::Account {
+            values,
+            randomness,
+            issuer: IpIdentity::from(4u32),
+        };
+        let commitment_inputs = vec![secrets.into()];
+        let proof = create_web3_id_proof(Web3IdProofInput {
+            request,
+            global_context,
+            commitment_inputs,
+        });
+
+        let data = serde_json::to_string_pretty(&proof?)?;
+        assert!(
+            serde_json::from_str::<Presentation<constants::ArCurve, Web3IdAttribute>>(&data)
+                .is_ok(),
+            "Cannot deserialize proof correctly."
+        );
+        Ok(())
+    }
```

Comment:
I feel that these tests don't really add value.

The functions are so trivial that this is just duplicating the tests that are there in the web3id modules.

Reply:
Yes, I guess I kinda just added it to have a test :/
I have changed it to parse from json, to have an example json input :thinking: 
If you still think it's "worthless" then I'll remove it :upside_down_face:

Reply:
No, the parsing test is good.